packages/editor-ui/src/stores/ui.store.ts

```diff
@@ -749,33 +765,43 @@ export const listenForModalChanges = (opts: {
 	});
 };
 
-export const generateUpgradeLinkUrl = async (
+export const generateUpgradeLink = async (
 	source: string,
 	utm_campaign: string,
 	deploymentType: string,
 ) => {
-	let linkUrl = '';
-
-	const searchParams = new URLSearchParams();
-
-	const cloudPlanStore = useCloudPlanStore();
-
+	let upgradeLink = N8N_PRICING_PAGE_URL;
 	if (deploymentType === 'cloud' && hasPermission(['instanceOwner'])) {
-		const adminPanelHost = new URL(window.location.href).host.split('.').slice(1).join('.');
-		const { code } = await cloudPlanStore.getAutoLoginCode();
-		linkUrl = `https://${adminPanelHost}/login`;
-		searchParams.set('code', code);
-		searchParams.set('returnPath', '/account/change-plan');
-	} else {
-		linkUrl = N8N_PRICING_PAGE_URL;
+		upgradeLink = await generateCloudDashboardAutoLoginLink({
+			redirectionPath: '/account/change-plan',
+		});
 	}
 
+	const url = new URL(upgradeLink);
+
 	if (utm_campaign) {
-		searchParams.set('utm_campaign', utm_campaign);
+		url.searchParams.set('utm_campaign', utm_campaign);
 	}
 
 	if (source) {
-		searchParams.set('source', source);
+		url.searchParams.set('source', source);
 	}
+
+	return url.toString();
+};
+
+const generateCloudDashboardAutoLoginLink = async (data: {
```

Comment:
nice refactor. Could we also refactor `redirectToDashboard` in cloud plans store to use this?